platform/broadcom/sonic-platform-modules-cel/dx010/modules/dx010_cpld.c

```diff
@@ -106,6 +114,53 @@ struct dx010_cpld_data {
 
 struct dx010_cpld_data *cpld_data;
 
+static ssize_t get_reset(struct device *dev, struct device_attribute *devattr,
+                char *buf)
+{
+        unsigned long reset = 0;
+
+        mutex_lock(&cpld_data->cpld_lock);
+
+        reset =
+                (inb(RESET3032) & 0x07) << (24+5) |
+                inb(RESET2229) << (24-3)  |
+                (inb(RESET1921) & 0x07) << (16 + 2) |
+                inb(RESET1118) << (16-6) |
+                (inb(RESET0910) & 0x03 ) << 8 |
+                inb(RESET0108);
+
+        mutex_unlock(&cpld_data->cpld_lock);
+
+        return sprintf(buf,"0x%8.8lx\n", reset & 0xffffffff);
+}
+
+static ssize_t set_reset(struct device *dev, struct device_attribute *devattr,
+                const char *buf, size_t count)
+{
+        unsigned long reset;
+        int err;
+
+        mutex_lock(&cpld_data->cpld_lock);
+
+        err = kstrtoul(buf, 16, &reset);
+        if (err)
+        {
+                mutex_unlock(&cpld_data->cpld_lock);
+                return err;
+        }
+
+        outb( (reset >> 0)  & 0xFF, RESET0108);
+        outb( (reset >> 8)  & 0x03, RESET0910);
+        outb( (reset >> 10) & 0xFF, RESET1118);
+        outb( (reset >> 18) & 0x07, RESET1921);
+        outb( (reset >> 21) & 0xFF, RESET2229);
+        outb( (reset >> 29) & 0x07, RESET3032);
```

Comment:
here it is better to check if it is zero or not, if none of the pin is put into reset mode, then you should not write to the register.

Reply:
total 32 pins, what about the sfp port?